spec/services/settings/working_days_update_service_spec.rb

```diff
@@ -0,0 +1,162 @@
+#
+#  OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+#  Copyright (C) 2006-2013 Jean-Philippe Lang
+#  Copyright (C) 2010-2013 the ChiliProject Team
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#  See COPYRIGHT and LICENSE files for more details.
+
+require 'spec_helper'
+require_relative 'shared/shared_call_examples'
+
+describe Settings::WorkingDaysUpdateService do
+  let(:instance) do
+    described_class.new(user:)
+  end
+  let(:user) { build_stubbed(:user) }
+  let(:contract) do
+    instance_double(Settings::UpdateContract,
+                    validate: contract_success,
+                    errors: instance_double(ActiveModel::Error))
+  end
+  let(:contract_success) { true }
+  let(:params_contract) do
+    instance_double(Settings::WorkingDaysParamsContract,
+                    valid?: params_contract_success,
+                    errors: instance_double(ActiveModel::Error))
+  end
+  let(:params_contract_success) { true }
+  let(:setting_name) { :a_setting_name }
+  let(:new_setting_value) { 'a_new_setting_value' }
+  let(:previous_setting_value) { 'the_previous_setting_value' }
+  let(:setting_params) { { setting_name => new_setting_value } }
+  let(:non_working_days_params) { {} }
+  let(:params) { setting_params.merge(non_working_days: non_working_days_params) }
+
+  before do
+    # stub a setting definition
+    allow(Setting)
+      .to receive(:[])
+          .and_call_original
+    allow(Setting)
+      .to receive(:[]).with(setting_name)
+          .and_return(previous_setting_value)
+    allow(Setting)
+      .to receive(:[]=)
+
+    # stub contract
+    allow(Settings::UpdateContract)
+      .to receive(:new)
+          .and_return(contract)
+    allow(Settings::WorkingDaysParamsContract)
+      .to receive(:new)
+          .and_return(params_contract)
+  end
+
+  describe '#call' do
+    subject { instance.call(params) }
+
+    shared_examples 'unsuccessful working days settings call' do
+      include_examples 'unsuccessful call'
+
+      context 'when non working days are present' do
+        include_examples 'unsuccessful call'
+
+        it 'does not persists the non working days' do
+          expect { subject }.not_to change(NonWorkingDay, :count)
+        end
+      end
```

Comment:
I'm not sure this nested context is needed. It looks like the shared examples `'unsuccessful call'` are called twice. Maybe the `it 'does not persist the non-working days'` should be moved one level up, and the `context 'when non working days are present'` removed.